application/src/main/java/org/thingsboard/server/actors/device/DeviceActorMessageProcessor.java

```diff
@@ -811,24 +811,28 @@ private void sendToTransport(ToServerRpcResponseMsg rpcMsg, UUID sessionId, Stri
     }
 
     private void saveRpcRequestToEdgeQueue(ToDeviceRpcRequest msg, Integer requestId) {
```

Comment:
Since the method is no longer blocking, you don't know if you saved event or not. You should pass the callback as the parameter of this method and not creating it inside the method. The caller of the method do not know if the event was actually saved.